### PR TITLE
Add contraint for compaction based indexed segments

### PR DESCRIPTION
**Pull request by xiaocai2333**
/kind improvement

### CODE REVIEW

**Comment by xiaocai2333** [comment]
rerun ut

File: internal/datacoord/garbage_collector.go [hunk 1]
```diff
@@ -241,9 +244,22 @@ func (gc *garbageCollector) clearEtcd() {
...
-	for _, segment := range drops {
+	dropIDs := lo.Keys(drops)
+	sort.Slice(dropIDs, func(i, j int) bool {
+		return dropIDs[i] < dropIDs[j]
+	})
+
+	for _, segmentID := range dropIDs {
+		segment, ok := drops[segmentID]
+		if !ok {
+			log.Error("segmentID is not in drops", zap.Int64("segmentID", segmentID))
```
Comment (czs007) [inline internal/datacoord/garbage_collector.go:255]: log.Warn

File: internal/util/paramtable/component_param.go [hunk 2]
```diff
@@ -1453,6 +1453,7 @@ type dataCoordConfig struct {
 	// compaction
 	EnableCompaction     bool
 	EnableAutoCompaction atomic.Value
+	BasedIndexedSegments atomic.Value
```
Comment (czs007) [inline internal/util/paramtable/component_param.go:1456]: IndexBasedCompaction

**Pull request merged**